components/ws-manager/pkg/manager/create.go

```diff
@@ -282,6 +282,22 @@ func (m *Manager) createDefiniteWorkspacePod(startContext *startWorkspaceContext
 		prefix = "ghost"
 	case api.WorkspaceType_IMAGEBUILD:
 		prefix = "imagebuild"
+		// mount self-signed gitpod CA certificate to ensure
+		// we can push images to the in-cluster registry
+		workspaceContainer.VolumeMounts = append(workspaceContainer.VolumeMounts,
+			corev1.VolumeMount{
+				Name:      "gitpod-ca-certificate",
+				MountPath: "/usr/local/share/ca-certificates/gitpod-ca.crt",
+				SubPath:   "ca.crt",
+				ReadOnly:  true,
+			},
+		)
+
+		// signal workspacekit to mount required files to push in-cluster registry
+		workspaceContainer.Env = append(workspaceContainer.Env, corev1.EnvVar{
+			Name:  "GITPOD_WORKSPACEKIT_BIND_MOUNTS",
+			Value: `["/usr/local/share/ca-certificates/gitpod-ca.crt"]`,
```

Comment:
workspacekit should detect this mount automatically because it's a secret. 
Does this branch work without this addition?